tests/unit/test_charm.py

```diff
@@ -428,11 +452,230 @@ def test_add_mongodb_org_repository_already_added_skips(
 
     @patch_network_get(private_address="1.1.1.1")
     def test_unit_ips(self):
-        key_values = {"private_address": "127.4.5.6"}
+        key_values = {"private-address": "127.4.5.6"}
         rel_id = self.harness.charm.model.get_relation("mongodb").id
         self.harness.add_relation_unit(rel_id, "mongodb/1")
         self.harness.update_relation_data(rel_id, "mongodb/1", key_values)
 
         resulting_ips = self.harness.charm._unit_ips
         expected_ips = ["127.4.5.6", "1.1.1.1"]
         self.assertEqual(resulting_ips, expected_ips)
+
+    @mock.patch("charm.MongodbOperatorCharm.app")
+    def test_mongodb_relation_joined_non_leader_does_nothing(self, app):
+        rel = self.harness.charm.model.get_relation("mongodb")
+        self.harness.set_leader(False)
+        self.harness.charm.on.mongodb_relation_joined.emit(relation=rel)
+
+        app.planned_units.assert_not_called()
+
+    @mock.patch("charm.MongodbOperatorCharm.app")
+    def test_mongodb_relation_joined_planned_units_does_not_match_joined_units(self, app):
+        rel = self.harness.charm.model.get_relation("mongodb")
+        self.harness.set_leader(True)
+        app.planned_units.return_value = 3
+
+        with self.assertLogs("charm", "DEBUG") as logs:
+            self.harness.charm.on.mongodb_relation_joined.emit(relation=rel)
+            self.harness.charm._add_mongodb_org_repository()
```

Comment:
Is this call really needed at this test? It looks like it isn't, but I'm not sure.